app/Jobs/TranscodeVideo.php

```diff
@@ -122,17 +122,9 @@ public function failed(?Throwable $exception): void
         // This directory stores local temp derivatives in case cloud storage is used.
         $localDisk->deleteDirectory($this->getTempDerivativesDirectoryPath());
 
-        if (!$this->oldVersionNumber) {
-            // A failed upload must not create a version.
-            $this->version->delete();
-            $versionNumber = $this->version->number - 1;
-        } else {
-            // Restoring an old version has failed, therefore we restore its previous state.
-            $this->version->update(['number' => $this->oldVersionNumber, 'processed' => $this->wasProcessed]);
-            $versionNumber = $this->oldVersionNumber;
-        }
+        $this->version->delete();
 
-        Transcode::callback($this->responseState ?? ResponseState::TRANSCODING_FAILED, $this->uploadToken, $this->version->Media, $versionNumber);
+        Transcode::callback($this->responseState ?? ResponseState::TRANSCODING_FAILED, $this->uploadToken, $this->version->Media, $this->oldVersionNumber ?? $this->version->number - 1);
```

Comment:
what if the version "number -1" does not exist? because it got deleted itself due to an error or similar?

maybe retrieve the latest processed version instead or something.

Reply:
I removed the old version number altogether and am now retrieving the latest processed version.